huma_test.go

```diff
@@ -806,6 +814,28 @@ func TestFeatures(t *testing.T) {
 			URL:    "/round-trip",
 			Body:   `{"$schema": "...", "name": "foo"}`,
 		},
+		{
+			Name: "recursive schema",
```

Comment:
I think there is also a need to check how the schema is created by creating a test case in the `TestSchema` test! Could you please create one? It is in the `schema_test.go` file.